### PR TITLE
[GITHUB-3] Update Role to New Standards

### PR DESCRIPTION
**Pull request by dennisconrad**
* Migrate to Molecule
* Add LICENSE
* Add "purge old kernels" functionality
* Bring other files up to date

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1 +1,3 @@
 ---
+
+security_hardening_purge_old_kernels: no
```
Comment (lobsterdore) [inline defaults/main.yml]: Var should be ```sansible_security_hardening_purge_old_kernels```

File: tasks/security_updates.yml [hunk 2]
```diff
@@ -1,11 +1,67 @@
...
-- name: Ensures latest Security updates are installed
-  become: yes
-  shell: "{{ item }}"
-  with_items:
-    - echo "# Running security udpate (`date`)" >> /var/log/apt-security-updates
-    - apt-get update >> /var/log/apt-security-updates
-    - echo "\n# Installing updates" >> /var/log/apt-security-updates
-    - DEBIAN_FRONTEND=noninteractive apt-get upgrade -o Aptitude::Delete-Unused=false --assume-yes --target-release `lsb_release -cs`-security >> /var/log/apt-security-updates
-    - echo "Security updates (if any) installed\n\n\n\n" >> /var/log/apt-security-updates
+- name: Install aptitude
```
Comment (lobsterdore) [inline tasks/security_updates.yml:3]: Needs become.

File: tasks/security_updates.yml [hunk 3]
```diff
@@ -1,11 +1,67 @@
...
+  apt:
+    cache_valid_time: 3600
+    name: aptitude
+
+- name: Log Update Start
```
Comment (lobsterdore) [inline tasks/security_updates.yml:9]: I think these log entries will need become as well.

File: tasks/security_updates.yml [hunk 4]
```diff
@@ -1,11 +1,67 @@
...
+  lineinfile:
+    create: yes
+    insertafter: EOF
+    line: "# Starting Security Update {{ ansible_date_time.iso8601 }}"
+    name: /var/log/apt-security-updates
+
+- name: Perform Update
```
Comment (lobsterdore) [inline tasks/security_updates.yml]: Needs become.

File: tasks/security_updates.yml [hunk 5]
```diff
@@ -1,11 +1,67 @@
...
+  apt:
+    cache_valid_time: 3600
+    default_release: "{{ ansible_distribution_release }}-security"
+    upgrade: yes
+  register: apt_upgrade
+
+- name: Log Update Details
```
Comment (lobsterdore) [inline tasks/security_updates.yml]: Needs become.

File: tasks/security_updates.yml [hunk 6]
```diff
@@ -1,11 +1,67 @@
...
+      when: "ansible_virtualization_type != 'docker'"
+
+    - name: Wait for System to Come Up
+      ping:
+      register: ping
+      until: ping | success
+      retries: 20
+      delay: 10
+
+    - name: Copy linux-purge
```
Comment (lobsterdore) [inline tasks/security_updates.yml]: Needs ```become: yes```

File: tasks/security_updates.yml [hunk 7]
```diff
@@ -1,11 +1,73 @@
...
-    - echo "# Running security udpate (`date`)" >> /var/log/apt-security-updates
-    - apt-get update >> /var/log/apt-security-updates
-    - echo "\n# Installing updates" >> /var/log/apt-security-updates
-    - DEBIAN_FRONTEND=noninteractive apt-get upgrade -o Aptitude::Delete-Unused=false --assume-yes --target-release `lsb_release -cs`-security >> /var/log/apt-security-updates
-    - echo "Security updates (if any) installed\n\n\n\n" >> /var/log/apt-security-updates
+- name: Install aptitude
+  become: yes
+  apt:
+    cache_valid_time: 3600
+    name: aptitude
```
Comment (lobsterdore) [inline tasks/security_updates.yml:7]: Why is Aptitude still needed? Haven't we removed the need to this by switching to the apt module?

File: tasks/security_updates.yml [hunk 8]
```diff
@@ -1,11 +1,73 @@
...
+        dest: /usr/bin/linux-purge
+        owner: root
+        group: root
+        mode: 0755
+
+    - name: Purge Old Kernels
+      become: yes
+      command: /usr/bin/linux-purge -k 0 -y
+      changed_when: no
+  when: sansible_security_hardening_purge_old_kernels
```
Comment (lobsterdore) [inline tasks/security_updates.yml:66]: Shall we only run this when apt_upgrade.changed is true?

Comment (dennisconrad) [reply tasks/security_updates.yml:66]: I was pondering that, but there could be old kernels installed even if this particular update run didn't install any.  You could run `linux-purge` in simulation mode and parse the output to see if anything needs to be removed, but in the end, you don't really save a step through that.

Comment (lobsterdore) [reply tasks/security_updates.yml:66]: OK that's fine by me 👍

File: tasks/security_updates.yml (hunk 7 above)
Comment (dennisconrad) [reply tasks/security_updates.yml:7]: The apt module uses aptitude as the backend; a fallback to "pure" apt is available in Ansible 2.4+.

File: tasks/security_updates.yml [hunk 9]
```diff
@@ -1,11 +1,73 @@
...
+      pause:
+        minutes: 1
+      when: "ansible_virtualization_type != 'docker'"
+
+    - name: Wait for System to Come Up
+      ping:
+      register: ping
+      until: ping | success
+      retries: 20
+      delay: 10
```
Comment (lobsterdore) [inline tasks/security_updates.yml]: This fails if SSH isn't available yet, this has happened to me a few times when testing against AWS. I suggest either lengthening the wait time or putting in an actual raw SSH command.

**Comment by lobsterdore** [comment]
LGTM 👍

**Review by rsgoodall — APPROVED** [review]

**Pull request merged**